lib/suse/connect/api.rb

```diff
@@ -205,12 +205,21 @@ def system_activations(auth)
       # @param auth [String] authorization string which will be injected in 'Authorization' header in request.
       #   In this case we expect Base64 encoded string with login and password
       # @param [Array <Remote::Product>] a list of producs
+      # @param target_base_product [Remote::Product] (optional) a target base
+      #   product to upgrade to. Only used by the backend when kind is :offline. Defaults to nil.
+      # @param kind [Symbol] (optional) :online or :offline. It specifies whether
+      #   the online or the offline migrations are desired. Defaults to :online.
       #
       # @return [Array <Array <Hash>>] the list of possible upgrade paths for the given products,
       #   where each product is represented by a hash with identifier, version, arch and release_type
-      def system_migrations(auth, products)
+      def system_migrations(auth, products, kind: :online, target_base_product: nil)
```

Comment:
@wstephenson I made the `kind` parameter optional here, but mandatory in [client.rb](https://github.com/SUSE/connect/pull/322/files#diff-ce4a8c41d114461bdba2a49aef6b34a2L146). What do you think, should we keep things consistent and make it mandatory here as well?